lib/action_controller/serialization.rb

```diff
@@ -25,6 +25,12 @@ def get_serializer(resource, options = {})
           "Please pass 'adapter: false' or see ActiveSupport::SerializableResource#serialize"
         options[:adapter] = false
       end
+
+      if ActiveModel::Serializer.enabled_adapters_by_media_type && !options.fetch(:adapter, nil)
+        adapter = ActiveModel::Serializer::Adapter.by_request(request)
+        options[:adapter] ||= adapter if adapter
+      end
```

Comment:
Would be really great to merge https://github.com/rails-api/active_model_serializers/pull/1017 first to take advantage of the register/get interface